clients/src/main/java/org/apache/kafka/clients/Metadata.java

```diff
@@ -416,14 +344,6 @@ private void updatePartitionInfo(String topic,
                 PartitionInfo previousInfo = cache.cluster().partition(tp);
                 if (previousInfo != null) {
                     partitionInfoConsumer.accept(previousInfo);
-                } else {
```

Comment:
Yes, I think so. Since we are not doing the weird two-step metadata update anymore, the `containsTopic` check below would always be true. Then it didn't seem possible that we would have a previous leader epoch for the partition, but not have the `PartitionInfo`. Maybe you can double-check the reasoning though.

Reply:
Makes sense.